docs/extensions/storage/_index.md

```diff
@@ -8,39 +8,144 @@ geekdocFilePath: _index.md
+- a *symlink*, or
+- a *reference* which can point to a resource in another *storage provider*
+
+### References
+
+A [*reference*](https://cs3org.github.io/cs3apis/#cs3.storage.provider.v1beta1.Reference) is a logical concept. It identifies a *resource* and consists of either
+- a *path* based reference, used to identify a *resource* in the *namespace* of a *storage provider*. It must start with a `/`.
+- an [*id* based reference](https://cs3org.github.io/cs3apis/#cs3.storage.provider.v1beta1.ResourceId), uniquely identifying a *resounce* in the *namespace* of a *storage provider*. It consists of a `storage provider id` and an `opaque id`. The `storage provider id` must NOT start with a `/`.
+
+{{< hint info >}}
+The `/` is important because currenty the static *storage registry* uses a map to look up which *storage provider* is responsible for the resource. Paths must be prefixed with `/` so there can be no collisions between paths and storage provider ids in the same map.
+{{< /hint >}}
+
+{{< hint warning >}}
+**Proposed Change**
+
+A *reference* is a logical concept. It identifies a *resource* and consists of
+a *root* and a *path* relative to that *root*. 
+A *root* is a `storage space id` and a `logical uuid` of a resource inside the *storage space*:
+```
+<storages_pace>:<logical_uuid>:<relative_path>
+`------------root------------´ `----path-----´
+```
+Both, *root* and *path*, are optional. The `storage space id` should be a uuid to prevent references from breaking when a *user* or *storage space* gets renamed. These are all valid references:
+
+| name | description |
+|------|-------------|
+|`::/users/alice/projects/foo` | follow this path in the  global namespace (path based reference)|
+|`home-alice::projects/foo` | in the *storage space* `home-alice`, start at the *root*, follow `projects/foo` |
+|`home-alice:c3cf23bb-8f47-4719-a150-1d25a1f6fb56:foo` | in the *storage space* `home-alice`, start at the resource identified by the `logical uuid` `c3cf23bb-8f47-4719-a150-1d25a1f6fb56`, follow `projects/foo` |
+|`home-alice:56f7ceca-e7f8-4530-9a7a-fe4b7ec8089a:` | in the *storage space* `home-alice`, start at the resource identified by the `logical uuid` `56f7ceca-e7f8-4530-9a7a-fe4b7ec8089a`, no relativ path to follow |
+|`ee1687e5-ac7f-426d-a6c0-03fed91d5f62:56f7ceca-e7f8-4530-9a7a-fe4b7ec8089a:` | in the *storage space* `ee1687e5-ac7f-426d-a6c0-03fed91d5f62`, start at the resource identified by the `logical uuid` `56f7ceca-e7f8-4530-9a7a-fe4b7ec8089a`, no relativ path to follow (id based reference) |
+
+They all reference the same *resource* (if the `logical uuid` of `projects` is 
+`c3cf23bb-8f47-4719-a150-1d25a1f6fb56` and the `logical uuid` of `foo` is `56f7ceca-e7f8-4530-9a7a-fe4b7ec8089a`).
+To move *storage spaces* between *storage providers*, the `logical uuid` of a file must be settable by the system. This also allows importing file metadata during a migration and restoring correct file ids when restoring a backup.
```

Comment:
"settable" by which system? Does it need to be editable, too?

Plus "restoring from backup", perhaps?

Reply:
I reworded this. should be clearer now.